lib/galaxy/auth/__init__.py

```diff
@@ -102,11 +108,21 @@ def check_auto_registration(self, trans, login, password):
                         else:
                             break  # end for loop if we can't make a unique username
                     log.debug("Email: %s, auto-register with username: %s" % (auto_email, auto_username))
-                    return (string_as_bool(options.get('auto-register', False)), auto_email, auto_username)
+                    auth_return["auto_reg"] = string_as_bool(options.get('auto-register', False))
+                    auth_return["email"] = auto_email
+                    auth_return["username"] = auto_username
+                    auth_return["auto_create_roles"] = string_as_bool(options.get('auto-create-roles', False))
+                    auth_return["auto_create_groups"] = string_as_bool(options.get('auto-create-groups', False))
+                    auth_return["auto_assign_roles_to_groups_only"] = string_as_bool(
+                        options.get('auto-assign-roles-to-groups-only', False))
+
+                    if len(auth_results) == 4:
+                        auth_return["attributes"] = auth_results[3]
+                    return auth_return
                 elif auth_result is None:
                     log.debug("Email: %s, Username %s, stopping due to failed non-continue" % (auto_email, auto_username))
                     break  # end authentication (skip rest)
-        return (False, '', '')
+        return auth_return
```

Comment:
I worry that the growing complexity of this return value structure will cause some extra logic in callers (although I guess there's only one caller now?). Plus the `auth_results` from the provider could be troublesome for other plugins or future plugin development. Maybe a class that can be shared for both purposes should be used instead?